plugins/modules/accesspoint_workflow_manager.py

```diff
@@ -0,0 +1,818 @@
+                return newdict
+            else:
+                self.log('Playbook AP Configuration remain same in Current AP configration',
+                      "INFO")
+                return None
+
+    def update_ap_configuration(self, device_data):
+        """
+        This function used to update the ap detail with the current ap configuration
+        Final data received from compare_ap_cofig_with_inputdata response will be the 
+        input of this function.
+        Parameters:
+          - device_data: DNAC final device data response from compare_ap_cofig_with_inputdata
+        Returns:
+            {
+                "response": {
+                    "taskId": "string",
+                    "url": "string"
+                },
+                "version": "string"
+            }
+        Example:
+            functions = Accesspoint(module)
+            final_input_data = functions.update_ap_configuration(device_data)
+        """
+
+        device = device_data
+        try:
+            self.log("Updating Access Point Configuration Information "+ device["macAddress"],
+                        "INFO")
+            # Below code might changed once we receive the dev dnac credentials
+            device["adminStatus"] = True
+            if device.get("apName") is not None:
+                device["apList"] = [dict(apName = device["apName"],
+                                    apNameNew = device["apNameNew"],
+                                    macAddress = device["macAddress"])]
+                del device["apName"]
+                del device["apNameNew"]
+            elif device.get("apName") is None and device.get("macAddress") is not None:
+                device["apList"] = [dict(macAddress = device["macAddress"])]
+
+            if device.get("location") is not None:
+                device["configureLocation"] = True
+            if device.get("ledBrightnessLevel") is not None:
+                device["configureLedBrightnessLevel"] = True
+            if device.get("ledStatus") is not None:
+                device["configureLedStatus"] = True
+                device["ledStatus"] = True if device["ledStatus"] == "Enabled" else False
+
+            for rmkey in ("mac_address", "hostname", "management_ip_address"):
+                if device.get(rmkey): del device[rmkey]
+            
+            #response = self.dnacsdk.wireless.configure_access_points(**device)
+            response = self.dnac._exec(
+                    family="wireless",
+                    function='configure_access_points',
+                    op_modifies=True,
+                    params=device,
+                )
+
+            self.log("Response of Access Point Configuration: " + str(response["response"]), "INFO")
+            return dict(macAdress=device["macAddress"], response=response["response"])
+
+        except Exception as e:
+            self.log("AP config update Error" + device["macAddress"] + str(e), "ERROR")
```

Comment:
Use format and set the status as failed if it comes into exception and return from here itslef.

Reply:
updated to format